tests/models/test_default_evaluator.py

```diff
@@ -182,15 +318,74 @@ def test_multi_classifier_evaluation(multiclass_logistic_regressor_model_uri, ir
     }
 
 
-def test_bin_classifier_evaluation(binary_logistic_regressor_model_uri, breast_cancer_dataset):
+def test_multi_classifier_evaluation_disable_logging_metrics_and_artifacts(
```

Comment:
Correct my understanding if I'm wrong. The default evaluator now evaluates both the candidate and the baseline model at the same time, not logging metrics and artifacts for the baseline model. In that case, doesn't it make sense to get rid of these `disable_logging` tests? Instead, for each model type, pass in a baseline model as well and verify that the baseline model did not have any side-effects. Maybe then, you won't have to have the `evaluate_model_helper` anymore.

Reply:
It is possible to test it that way; but It is hard to verify if a baseline model has no side effect, especially in the case that a baseline model may have same metrics that have same value as candidate model; hence a direct way is just evaluate a model only with `is_baseline_model` flag set to True and see if there is nothing logged and no artifact generated.

Reply:
For example, the `example_count` should be the same for all models, so it is not possible to test if baseline model also logs `example_count` that overwrote candidate model's logged metrics, it is not likely that would happen; but to test it strictly I think we should do this way which is more deterministic as this method is agnostic to model.